modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/MimeTypeParametersDeclarationEnricher.java

```diff
@@ -0,0 +1,132 @@
+import static org.mule.runtime.module.extension.internal.util.IntrospectionUtils.isVoid;
+import static org.mule.runtime.module.extension.internal.util.MuleExtensionUtils.getImplementingMethod;
+import static org.mule.runtime.module.extension.internal.util.MuleExtensionUtils.getImplementingType;
+
+import org.mule.metadata.api.ClassTypeLoader;
+import org.mule.metadata.api.annotation.EnumAnnotation;
+import org.mule.metadata.api.model.BinaryType;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.model.StringType;
+import org.mule.metadata.api.visitor.MetadataTypeVisitor;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.OperationDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterGroupDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterizedDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.SourceDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.WithSourcesDeclaration;
+import org.mule.runtime.api.meta.model.display.LayoutModel;
+import org.mule.runtime.api.util.Reference;
+import org.mule.runtime.extension.api.declaration.fluent.util.IdempotentDeclarationWalker;
+import org.mule.runtime.extension.api.declaration.type.ExtensionsTypeLoaderFactory;
+import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+import org.mule.runtime.extension.api.runtime.source.Source;
+import org.mule.runtime.module.extension.internal.ExtensionProperties;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/**
+ * Enriches operations which return types are {@link InputStream}, {@link String} or {@link Object} by adding two parameters:
+ * A {@link ExtensionProperties#MIME_TYPE_PARAMETER_NAME} that allows configuring the mimeType to the output operation payload
+ * and a {@link ExtensionProperties#ENCODING_PARAMETER_NAME} that allows configuring the encoding to the output operation payload.
+ * <p>
+ * Both added attributes are optional without default value and accept expressions.
+ *
+ * @since 4.0
+ */
+public final class MimeTypeParametersDeclarationEnricher extends AbstractAnnotatedDeclarationEnricher {
+
+  private final ClassTypeLoader typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
+
+  @Override
+  public void enrich(ExtensionLoadingContext extensionLoadingContext) {
+    final ExtensionDeclaration declaration = extensionLoadingContext.getExtensionDeclarer().getDeclaration();
+    new IdempotentDeclarationWalker() {
+
+      @Override
+      protected void onOperation(OperationDeclaration declaration) {
+        getImplementingMethod(declaration).ifPresent(method -> {
```

Comment:
You should not need the implementing method for this. In the metadata type you have all the necessary information to do this. This enricher should be in the ExtensionFactory, not a custom one